src/test/addrman_tests.cpp

```diff
@@ -49,17 +57,11 @@ static std::vector<bool> FromBytes(const unsigned char* source, int vector_size)
     return result;
 }
 
-/* Utility function to create a deterministic addrman, as used in most tests */
-static std::unique_ptr<AddrMan> TestAddrMan(std::vector<bool> asmap = std::vector<bool>())
```

Comment:
This utility function was suggested to me by @jnewbery in https://github.com/bitcoin/bitcoin/pull/23826#discussion_r773348729 (Github link doesn't really work for me but I just unresolved that conversation). Since it is removed again - do you disagree with that? Otherwise, you could have just adjusted that function to use `DETERMINISTIC` and `GetCheckRatio(m_node)`.

Reply:
> This line appears 17 times in the test. It's also very noisy and distracting from the purpose of the test:


```cpp
auto addrman = std::make_unique<AddrMan>(std::vector<bool>(), /*deterministic=*/true, /*consistency_check_ratio=*/100);
```

I agree. In `master` the function `TestAddrMan()` makes sense IMO.

However, with this PR, we also have to pass the ratio to it (or `m_node` so that it can call `GetCheckRatio()`) and then in some places addrman is constructed with `deterministic=false` and for this `TestAddrMan()` is not used which is not consistent/uniform way of creating addrman objects across the file. So, to get it all consistent and still use `TestAddrMan()` it would have to be extended to take a boolean argument (deterministic) and the ratio or `NodeContext` object, so calls would look like:

```cpp
auto addrman = TestAddrMan(EMPTY_ASMAP, DETERMINISTIC, GetCheckRatio(m_node));
// or
auto addrman = TestAddrMan(EMPTY_ASMAP, DETERMINISTIC, m_node);
```

which I think is an unnecessary indirection and better construct the object directly:

```cpp
auto addrman = std::make_unique<AddrMan>(EMPTY_ASMAP, DETERMINISTIC, GetCheckRatio(m_node));
```

Anyway, I would restore `TestAddrMan()` if you prefer it over direct construction. Just suggest exactly what arguments should it take and whether it should be used on places where addrman objects are created on-stack, without `unique_ptr`.

Reply:
Ah, I hadn't realized that we would have to pass `m_node` each time now.
I agree with you that the explicit construction makes more sense in this case - it was mostly for better readability, and the more arguments we'd have to pass each time, the less sense it makes. Also, making the calls shorter with the constants EMPTY_ASMAP, DETERMINISTIC instead of the long named arguments each time is an alternative way to make the line more readable.

So, I think it makes sense the way it is.

Reply:
Yeah, my thought exactly! You said it with less words :)